### PR TITLE
【fleet】change to fleet Model

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### CODE REVIEW

File: paddleformers/transformers/auto/modeling.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def _get_model_class_from_config(cls, pretrained_model_name_or_path, config_file
             model_class = getattr(import_class, init_class)
             return model_class
         except AttributeError:
-            model_class = getattr(import_class, init_class + "Fleet")
+            model_class = getattr(import_class, init_class + "Decapitate")
```
Comment (From00) [inline paddleformers/transformers/auto/modeling.py]: Deprecated

File: tests/config/ci/glm45_dpo.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ max_steps: 10
 eval_steps: 100000
 evaluation_strategy: steps
 #pipeline_model_parallel_size: 1
-continue_training: false
+continue_training: true
```
Comment (changeyoung98) [inline tests/config/ci/glm45_dpo.yaml:37]: 原yaml continue training为false，进入workflow冷启动分支，from_config调用两次，第一次is_lora为false，构建Fleet模型，第二次is_lora为true，fallback到formers模型，CI报错。

Image

File: tests/config/ci/glm45_lora.yaml [hunk 3]
```diff
@@ -64,4 +64,4 @@ load_checkpoint_format: flex_checkpoint
 save_checkpoint_format: flex_checkpoint
 
 moe_router_force_load_balancing: true
-num_empty_layers_add_in_head: 1
\ No newline at end of file
+num_empty_layers_add_in_tail: 1
```
Comment (changeyoung98) [inline tests/config/ci/glm45_lora.yaml:67]: Formers组网无法识别头部空层，aoa可以识别，会导致aoa映射错层，热启失败。

File: examples/config/dpo/full_function_call.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ split_multi_turn: False
 ### model
 model_name_or_path: Qwen/Qwen3-0.6B-Base
 attn_impl: flashmask
-use_fused_head_and_loss_fn: true
+use_fused_head_and_loss_fn: false
```
Comment (xiaoguoguo626807) [inline examples/config/dpo/full_function_call.yaml:18]: 此开关控制lm_head和loss的fuse, fleet lm_head 暂时不支持

File: paddleformers/transformers/auto/modeling.py [hunk 5]
```diff
@@ -195,6 +195,9 @@ def _get_model_class_from_config(cls, pretrained_model_name_or_path, config_file
         import_class = importlib.import_module(f"paddleformers.transformers.{class_name}.modeling")
         if is_lora:
             try:
+                model_class = getattr(import_class, init_class + "Decapitated")
+                return model_class
+            except AttributeError:
```
Comment (xiaoguoguo626807) [inline paddleformers/transformers/auto/modeling.py:200]: lora 情况暂时回退到旧模型

File: tests/config/ci/qwen3_multicard_lora.yaml [hunk 6]
```diff
@@ -73,6 +73,7 @@ fuse_swiglu: true
 fuse_rms_norm: true
 fuse_linear: true
 use_paddlefleet: true
+use_qk_norm: true
```
Comment (changeyoung98) [inline tests/config/ci/qwen3_multicard_lora.yaml:76]: formers组网默认开启use_qk_norm，fleet组网需要手动开启，否则SFT无法保存q_norm

File: scripts/regression/test_dpo_tiny-random-glm4moe.py [hunk 7]
```diff
@@ -35,15 +35,15 @@
...
+DPO_FULL_RESUME_EXCEPTED_LOSS = 0.693408
 DPO_FULL_EXCEPTED_RESULT = [[10564, 10564, 102954, 47231, 47231, 47231, 47231, 47231, 47231, 47231]]
 
 DPO_LORA_EXCEPTED_LOSS = 0.693147
 DPO_LORA_RESUME_EXCEPTED_LOSS = 0.692793
 DPO_LORA_EXCEPTED_RESULT = [[51172, 37927, 96130, 27654, 133362, 95331, 27654, 133362, 115845, 115845]]
 
 DPO_FULL_TP_PP_EXCEPTED_LOSS = 0.693147
-DPO_FULL_TP_PP_RESUME_EXCEPTED_LOSS = 0.694038
+DPO_FULL_TP_PP_RESUME_EXCEPTED_LOSS = 0.693166
```
Comment (xiaoguoguo626807) [inline scripts/regression/test_dpo_tiny-random-glm4moe.py:46]: 切换模型导致的精度diff

File: scripts/regression/test_dpo_tiny-random-glm4moe.py [hunk 8]
```diff
@@ -101,11 +101,13 @@ def create_and_check_model_generate(
...
     ):
-        from paddleformers.transformers import Glm4MoeForCausalLM
+        from paddleformers.transformers.glm4_moe.modeling import (
+            Glm4MoeForCausalLMDecapitated,
+        )
 
         input_ids = paddle.to_tensor([[1, 306, 4658, 278, 6593, 310, 2834, 338]])
         attention_mask = paddle.ones_like(input_ids)
-        model = Glm4MoeForCausalLM.from_pretrained(model_path, dtype="bfloat16", convert_from_hf=True)
+        model = Glm4MoeForCausalLMDecapitated.from_pretrained(model_path, dtype="bfloat16", convert_from_hf=True)
```
Comment (xiaoguoguo626807) [inline scripts/regression/test_dpo_tiny-random-glm4moe.py:110]: fleet 暂时没有支持generate

File: tests/config/ci/qwen3_multicard_lora.yaml [hunk 9]
```diff
@@ -50,7 +50,7 @@ learning_rate: 1.0e-5
 
 # performance
 recompute_granularity: ""
-apply_rope_fusion: true
+# apply_rope_fusion: true
```
Comment (xiaoguoguo626807) [inline tests/config/ci/qwen3_multicard_lora.yaml:53]: formers 模型开启这个flag 会报错

File: tests/integration_test/glm45_dpo.sh [hunk 10]
```diff
@@ -32,7 +32,7 @@ config_json=$CACHE_DIR/glm45/GLM-4.5-Air/config.json
 
 yq '.train_dataset_path = strenv(cur_dir) + "/dpo_data/dpo_train.jsonl"
     | .eval_dataset_path = strenv(cur_dir) + "/dpo_data/dpo_eval.jsonl"
-    | .model_name_or_path = strenv(cur_dir) + "/checkpoints/glm_single_lora_ckps"
+    | .model_name_or_path = strenv(cur_dir) + "/checkpoints/glm_full_pp_ckpts"
     | .logging_dir = strenv(cur_dir) + "/glm_full_dpo_vdl_log"
```
Comment (xiaoguoguo626807) [inline tests/integration_test/glm45_dpo.sh:36]: 之前没有load 权重， 监控的冷启动， dpo 不能load lora 权重

File: tests/transformers/glm4_moe/test_modeling.py [hunk 11]
```diff
@@ -277,17 +293,16 @@ def create_and_check_gqa_model(self, config, input_ids, input_mask, *args):
...
+        data = {
+            "input_ids": input_ids,
+            "position_ids": None,
+            "attention_mask": None,
+            "labels": input_ids if self.parent.use_labels else None,
+        }
+
+        result = model(data)
+
+        self.parent.assertEqual(result.shape, [self.batch_size, self.seq_length, self.vocab_size])
```
Comment (hushenwei2000) [inline tests/transformers/glm4_moe/test_modeling.py]: Formers 模型的 forward() 如果传入了 labels 那么返回 list: [loss, hidden_states]，否则那么返回 list [hidden_states]。
Fleet 模型的 forward() 只返回 hidden_states

File: tests/transformers/glm4_moe/test_modeling.py [hunk 12]
```diff
@@ -237,35 +248,40 @@ def create_and_check_lm_head_model(self, config, input_ids, input_mask, *args):
...
-            input_ids,
-            labels=input_ids if self.parent.use_labels else None,
-            return_dict=self.parent.return_dict,
-        )
+        data = {
+            "input_ids": input_ids,
+            "position_ids": None,
+            "attention_mask": None,
+            "labels": input_ids if self.parent.use_labels else None,
+        }
```
Comment (xiaoguoguo626807) [inline tests/transformers/glm4_moe/test_modeling.py:271]: fleet 模型输入格式为字典

File: tests/transformers/glm4_moe/test_modeling.py [hunk 13]
```diff
@@ -371,43 +424,6 @@ def test_determinism(self):
...
-
-                for k, v in model_state_3.items():
-                    md53 = v._md5sum()
-                    md52 = model_state_2[k]._md5sum()
-                    if k.endswith(".mlp.gate.weight"):
-                        md52 = model_state_2[k].cast("bfloat16")._md5sum()
-                        md53 = model_state_3[k].cast("bfloat16")._md5sum()
-                    print(k)
-                    assert md52 == md53
-
```
Comment (xiaoguoguo626807) [inline tests/transformers/glm4_moe/test_modeling.py:410]: 测试uc 和 fc 的对比， fleet 模型没有适配uc 因此删除

File: tests/transformers/qwen3moe/test_modeling.py [hunk 14]
```diff
@@ -208,9 +211,17 @@ def create_and_check_for_causal_lm(
         choice_labels,
     ):
         model = Qwen3MoeForCausalLM(config=config)
-        model.eval()
-        result = model(input_ids, attention_mask=input_mask, labels=token_labels, return_dict=True)
-        self.parent.assertEqual(result.logits.shape, [self.batch_size, self.seq_length, self.vocab_size])
+        model = paddle.amp.decorate(models=model, level="O2", dtype="bfloat16")
+        data = {
```
Comment (xiaoguoguo626807) [inline tests/transformers/qwen3moe/test_modeling.py:215]: fused 算子只支持bfloat16

File: tests/transformers/qwen3moe/test_modeling.py [hunk 15]
```diff
@@ -442,39 +537,23 @@ def test_Qwen3Moe_converter_from_local_dir(self):
             # 2. forward the paddle model with fc
             from paddleformers.transformers import Qwen3MoeConfig, Qwen3MoeForCausalLM
 
-            paddle_model = Qwen3MoeForCausalLM.from_pretrained(
```
Comment (xiaoguoguo626807) [inline tests/transformers/qwen3moe/test_modeling.py:445]: fleet 模型只有fused 路径

**Review by lugimzzz — DISMISSED** [review]
lgtm
